src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -147,6 +148,14 @@ private void updateActionForDeployment(Map<String, Object> newConfig, Deployment
             return;
         }
 
+        boolean reconnected = connectivityValidator
+                .validateConnectivity(totallyCompleteFuture, nucleusConfig, deployment);
+        connectivityValidator.close();
```

Comment:
calling close here seems like a smell. `validateConnectivity` should close anything that it needs to close.